sql_assignment4_queries.sql

```diff
@@ -0,0 +1,340 @@
+
+CREATE TABLE locations (
+  LOCATION_ID decimal(4,0) NOT NULL DEFAULT '0',
+  STREET_ADDRESS varchar(40) DEFAULT NULL,
+  POSTAL_CODE varchar(12) DEFAULT NULL,
+  CITY varchar(30) NOT NULL,
+  STATE_PROVINCE varchar(25) DEFAULT NULL,
+  COUNTRY_ID varchar(2) DEFAULT NULL,
+  PRIMARY KEY (LOCATION_ID),
+  ) 
+ 
+INSERT INTO locations (LOCATION_ID, STREET_ADDRESS, POSTAL_CODE, CITY, STATE_PROVINCE, COUNTRY_ID) VALUES
+('1000', '1297 Via Cola di Rie', '989', 'Roma', '', 'IT'),
+('1100', '93091 Calle della Testa', '10934', 'Venice', '', 'IT'),
+('1200', '2017 Shinjuku-ku', '1689', 'Tokyo', 'Tokyo Prefecture', 'JP'),
+('1300', '9450 Kamiya-cho', '6823', 'Hiroshima', '', 'JP'),
+('1400', '2014 Jabberwocky Rd', '26192', 'Southlake', 'Texas', 'US'),
+('1500', '2011 Interiors Blvd', '99236', 'South San Francisco', 'California', 'US'),
+('1600', '2007 Zagora St', '50090', 'South Brunswick', 'New Jersey', 'US'),
+('1700', '2004 Charade Rd', '98199', 'Seattle', 'Washington', 'US'),
+('1800', '147 Spadina Ave', 'M5V 2L7', 'Toronto', 'Ontario', 'CA'),
+('1900', '6092 Boxwood St', 'YSW 9T2', 'Whitehorse', 'Yukon', 'CA'),
+('2000', '40-5-12 Laogianggen', '190518', 'Beijing', '', 'CN'),
+('2100', '1298 Vileparle (E)', '490231', 'Bombay', 'Maharashtra', 'IN'),
+('2200', '12-98 Victoria Street', '2901', 'Sydney', 'New South Wales', 'AU'),
+('2300', '198 Clementi North', '540198', 'Singapore', '', 'SG'),
+('2400', '8204 Arthur St', '', 'London', '', 'UK'),
+('2500', '"Magdalen Centre', ' The Oxford ', 'OX9 9ZB', 'Oxford', 'Ox'),
+('2600', '9702 Chester Road', '9629850293', 'Stretford', 'Manchester', 'UK'),
+('2700', 'Schwanthalerstr. 7031', '80925', 'Munich', 'Bavaria', 'DE'),
+('2800', 'Rua Frei Caneca 1360', '01307-002', 'Sao Paulo', 'Sao Paulo', 'BR'),
+('2900', '20 Rue des Corps-Saints', '1730', 'Geneva', 'Geneve', 'CH'),
+('3000', 'Murtenstrasse 921', '3095', 'Bern', 'BE', 'CH'),
+('3100', 'Pieter Breughelstraat 837', '3029SK', 'Utrecht', 'Utrecht', 'NL'),
+('3200', 'Mariano Escobedo 9991', '11932', 'Mexico City', '"Distrito Federal', '"');
+
+CREATE TABLE regions (
+  REGION_ID decimal(5,0) NOT NULL,
+  REGION_NAME varchar(25) DEFAULT NULL UNIQUE,
+  PRIMARY KEY (REGION_ID)
+)
+INSERT INTO regions (REGION_ID, REGION_NAME) VALUES
+('1', 'Europe'),
+('2', 'Americas'),
+('3', 'Asia'),
+('4', 'Middle East and Africa');
+
+/****************************************************** SQL Assignments 4**********************************************/
+
+/*Write a query to get unique department ID from employee table.*/
+select distinct(DEPARTMENT_ID) from employees;
+
+/*Write a query to get all employee details from the employee table order by first name, descending.*/
+select * from employees order by FIRST_NAME DESC;
+
+/*Write a query to get the employee ID, names (first_name, last_name), salary in ascending order of
+salary.*/
+select EMPLOYEE_ID, FIRST_NAME, LAST_NAME, SALARY from employees order by salary;
```

Comment:
The question mentions names (first_name, last_name). Can you try modifying the query ?

Reply:
Removed employee_id from query